web-components/collapse/collapse.svelte

```diff
@@ -0,0 +1,131 @@
+<svelte:options tag="leo-collapse" />
+
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte'
+
+  export let title = ''
+  export let isOpen: boolean | undefined
+  $: isOpenInternal = isOpen ?? false
+
+  const dispatcher = createEventDispatcher()
+  const toggle = (e) => {
+    e.preventDefault()
+
+    // We only support toggling if this component isn't being 'controlled'
+    const toggleTo = !isOpenInternal
+    if (isOpen === undefined) isOpenInternal = toggleTo
+    dispatcher('toggle', { open: toggleTo })
+  }
+</script>
+
+<details class="leoCollapse" open={isOpenInternal}>
+  <summary on:click={toggle}>
+    {#if $$slots.icon}
+      <div class="icon">
+        <slot name="icon" />
```

Comment:
Do we want a default icon here?

Reply:
I was thinking it's probably fine to have no icon by default, but maybe we want to ensure all collapses have a default icon, not sure.